Further fetchone() performance improvement

Lordshinjo opened the pull request:
- In `DataRow` message, use slicing on the `data` bytearray rather than
`unpack_from`, which saves from having to create the format string
- The value returned is now a `bytearray`, which means that the `compat`
module needs to handle `bytearray` in addition to `bytes`, considering them
as equivalent

sitingren commented:
Do you have any benchmark for this improvement?

sitingren commented:
This improvement is a user visible change. The query result value can change from `bytes` to `bytearray` for some [SQL types](https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/column.py#L138). Do you think other users may complain about this change in any situation?

Lordshinjo commented:
> Do you have any benchmark for this improvement?

Redoing a similar benchmark as in #356 (`fetchall()` of 1M rows from a 100-columns tables):
* Before #356: 225s
* After #356: 200s
* After this: 155s

(the numbers are a bit different from the previous ones, probably because the data retrieved is a bit different - I added a `ORDER BY` for more stability of the retrieved results, so I'm probably fetching a different set of 1M rows)

> This improvement is a user visible change. The query result value can change from `bytes` to `bytearray` for some [SQL types](https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/column.py#L138). Do you think other users may complain about this change in any situation?

While `bytes` and `bytearray` are pretty much functionally identical in Python 3 (except for the mutability), changing from one to the other would probably still break something, because of potential `isinstance()` checks.
I reverted to the previous behavior.

sitingren commented:
I see. Looks like this improvement is significant. Thank you for making good enhancements!
